packages/wdio-browserstack-service/src/accessibility-handler.ts

```diff
@@ -216,8 +245,26 @@ class _AccessibilityHandler {
 
         try {
             const shouldScanScenario = shouldScanTestForAccessibility(featureData?.name, pickleData.name, this._accessibilityOptions, world, true)
+            if (isAppAccessibilityAutomationSession(this._accessibility, this.isAppAutomate)) {
+                this._testMetadata[uniqueId] = {
+                    scanTestForAccessibility : shouldScanScenario,
+                    accessibilityScanStarted : true
+                }
+                this._testMetadata[uniqueId].accessibilityScanStarted = shouldScanScenario
+                if (this._sessionId) {
+                    /* For case with multiple tests under one browser, before hook of 2nd test should change this map value */
+                    AccessibilityHandler._a11yScanSessionMap[this._sessionId] = shouldScanScenario
+                }
+                Listener.setTestRunAccessibilityVar(this._accessibility && shouldScanScenario)
+                if (shouldScanScenario) {
+                    BStackLogger.info('AppAutomate test case execution has started.')
+                }
+                return
+            }
             const isPageOpened = await this.checkIfPageOpened(this._browser, uniqueId, shouldScanScenario)
-
+            if (!isPageOpened) {
```

Comment:
reason for moving this?

Reply:
AppAutomate session doesn't need this pageOpened check its applicable only for automate session

Reply:
Ya this isn't removed. This is moved later in the code.